lib/opFns.js

```diff
@@ -765,28 +764,27 @@ function trap (err) {
 /**
  * Subtracts the amount needed for memory usage from `runState.gasLeft`
  * @method subMemUsage
- * @param {Number} offset
- * @param {Number} length
+ * @param {BN} offset
+ * @param {BN} length
  * @return {String}
  */
 function subMemUsage (runState, offset, length) {
   //  abort if no usage
-  if (!length) return
+  if (length.isZero()) return
 
   // hacky: if the dataOffset is larger than the largest safeInt then just
   // load 0's because if tx.data did have that amount of data then the fee
   // would be high than the maxGasLimit in the block
-  if (offset > MAX_INT || length > MAX_INT) {
+  if (offset.gt(MAX_INT) || length.gt(MAX_INT)) {
     runState.gasLeft = new BN(0)
     trap(ERROR.OUT_OF_GAS)
   }
 
-  const newMemoryWordCount = Math.ceil((offset + length) / 32)
-
-  if (newMemoryWordCount <= runState.memoryWordCount) return
+  const newMemoryWordCount = offset.add(length).divRound(new BN(32))
```

Comment:
Thanks for the review, but `divRound` is supposed to round up and also had an assertion for comparing the two versions. It might still be wrong, but most of the failures I've found weren't gas related, but plain crashes.

This PR is not finished and have found a lot of other bugs.